src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2761,24 +2771,32 @@ kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
           const char *arg = *iter;
           if (!g_strv_contains (existing_kargs, arg))
             {
+              changed = TRUE;
               ostree_kernel_args_append (kargs, arg);
             }
-
         }
     }
+
   if (delete_if_present)
     {
       for (char **iter = delete_if_present; iter && *iter; iter++)
         {
           const char *arg = *iter;
           if (g_strv_contains (existing_kargs, arg))
             {
+              changed = TRUE;
               if (!ostree_kernel_args_delete (kargs, arg, error))
                 return FALSE;
             }
         }
      }
 
+  if(!changed && !vardict_lookup_bool (self->options, "reboot", FALSE))
```

Comment:
There's no need to consult `reboot` here. If nothing changed, then we're not rebooting anyway. :)

Reply:
Also, looks like we're missing a space between `if` and the parenthesis.